File: go/internal/orchestrator/state_commit.go
package orchestrator

import (
	"fmt"
	"strings"

	"all-you-need-is-git/go/internal/gitx"
)

type stateTrailer struct {
	Key   string
	Value string
}

func buildStateCommitMessage(subject string, prompt string, trailers []stateTrailer) string {
	cleanSubject := strings.TrimRight(subject, "\n")
	cleanPrompt := strings.TrimRight(prompt, "\n")

	parts := []string{cleanSubject, "", cleanPrompt, ""}
	for _, trailer := range trailers {
		parts = append(parts, fmt.Sprintf("%s: %s", trailer.Key, trailer.Value))
	}
	return strings.Join(parts, "\n") + "\n"
}

func commitState(worktreePath string, subject string, prompt string, trailers []stateTrailer) error {
	message := buildStateCommitMessage(subject, prompt, trailers)
	return gitx.Commit(worktreePath, message, true)
}

File: js/AgentsOrchestrator/stateCommit.js
export function buildStateCommitMessage(subject, prompt, trailers) {
  const cleanSubject = String(subject || '').replace(/\n+$/, '');
  const cleanPrompt = String(prompt || '').replace(/\n+$/, '');

  const lines = [cleanSubject, '', cleanPrompt, ''];
  for (const trailer of trailers || []) {
    lines.push(`${trailer.key}: ${trailer.value}`);
  }

  return `${lines.join('\n')}\n`;
}

export async function commitState(worktreeGit, subject, prompt, trailers) {
  const message = buildStateCommitMessage(subject, prompt, trailers);
  await worktreeGit.commit(message, { '--allow-empty': null });
}
